src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -59,23 +59,24 @@ public class DocumentBuilder : IDisposable
                 {
                     _currentBuildInfo.PluginHash = ComputePluginHash(assemblyList);
                     _currentBuildInfo.TemplateHash = templateHash;
-                    _currentBuildInfo.DirectoryName = IncrementalUtility.CreateRandomDirectory(intermediateFolder);
+                    _currentBuildInfo.DirectoryName = IncrementalUtility.CreateRandomDirectory(Environment.ExpandEnvironmentVariables(intermediateFolder));
                 }
             }
             Logger.LogInfo($"{Processors.Count()} plug-in(s) loaded.");
             foreach (var processor in Processors)
             {
                 Logger.LogVerbose($"\t{processor.Name} with build steps ({string.Join(", ", from bs in processor.BuildSteps orderby bs.BuildOrder select bs.Name)})");
             }
-            if (intermediateFolder != null)
-            {
-                var expanded = Environment.ExpandEnvironmentVariables(intermediateFolder);
-                if (expanded.Length == 0)
-                {
-                    expanded = ".";
-                }
-                _intermediateFolder = Path.GetFullPath(expanded);
-            }
+            _intermediateFolder = intermediateFolder;
+            //if (intermediateFolder != null)
```

Comment:
removed.

---
In reply to: [102641811](https://github.com/dotnet/docfx/pull/1339#discussion_r102641811) [](ancestors = 102641811)